src/redux/reducers/main.ts

```diff
@@ -357,6 +357,9 @@ export const mainSlice = createSlice({
       if (!resource) {
         return;
       }
+
+      removeReferringRefs(resource, state.resourceMap);
```

Comment:
@devcatalin @erdkse not sure we need to call this method from elsewhere in other scenarios/actions!?

Reply:
this looks like the only place